### PR TITLE
Add UC Volumes local target directory as an MPD disable condition

### PR DESCRIPTION
**Pull request by BenWilson2**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Add to the MPD disablement logic for UC Volumes due to lack of support for seek writes. 

## How is this patch tested?

- [ ] Existing unit/integration tests
- [X] New unit/integration tests
- [X] Manual tests (describe details, including test results, below)

Validated that UC Volumes are case-sensitive (similar to FUSE mount points):

``` bash
%sh ls -arlt /Volumes/
```

```text
total 448
drwxr-xr-x 1 root   root    4096 Sep 13 20:07 ..
drwxrwxrwx 2 nobody nogroup 4096 Sep 13 20:07 .
...
```

```bash
%sh ls -arlt /volumes/
```

```text
ls: cannot access '/volumes/': No such file or directory
```

## Does this PR require documentation update?

- [X] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [X] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [X] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by BenWilson2** [comment]
cc @sbauersfeld as a follow up to #9545

**Review by sbauersfeld — APPROVED** [review]

File: mlflow/utils/uri.py [hunk 1]
```diff
@@ -72,11 +73,14 @@ def is_databricks_uri(uri):
...
-def is_fuse_uri(uri):
+def is_fuse_or_uc_volumes_uri(uri):
     """
-    Validates whether a provided URI is directed to a FUSE mount point
+    Validates whether a provided URI is directed to a FUSE mount point or a UC volumes mount point
     """
-    return any(uri.startswith(x) for x in [_DBFS_FUSE_PREFIX, _DBFS_HDFS_URI_PREFIX])
+    return any(
+        uri.startswith(x)
+        for x in [_DBFS_FUSE_PREFIX, _DBFS_HDFS_URI_PREFIX, _UC_VOLUMES_URI_PREFIX]
```
Comment (dbczumar) [inline mlflow/utils/uri.py:82]: What if there are multiple slashes?

Comment (dbczumar) [reply mlflow/utils/uri.py:82]: Are there any other ways to reference UC volumes?

Comment (BenWilson2) [reply mlflow/utils/uri.py]: 1. Awesome catch on the multiple slashes. TY!! 
2. Yep! There's an internal one that customers (shouldn't use) but I'm adding it in for safety anyway. It'll be retired in the next few quarters, but in the meantime, it's still a failure mechanism, so I'm adding in an exclusion for it.

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @BenWilson2 !

**Pull request merged**